### DISCUSSION TITLE
Display labels of selected items in filtered Multiselect component

### DISCUSSION THREAD

**Discussion by mohamadbdeir** [primefaces/community#88] [PrimeReact]
I was trying to use a filtered multiselect component with the data being fetched from an API instead of hardcoded.
 In my code, typing in the filter field performs an API call and fetches data to be displayed in the options.
However, I have noticed that when the options are filtered, if a selected option is filtered out so no longer available in the list of options to select, the label is showing 'null' instead of the object field. I came to the conclusion that the labels are being fetched from the list of the options instead of the list of selected items.
I think it would be better to display labels based on the list of already selected items.

![opera_HldnZ3GoaN](https://github.com/primefaces/community/assets/58555851/784a00ec-cffa-4fda-b7b3-ba6f863ab263)

**Comment by melloware** [comment]
If you can create a StackBlitz reproducer then please open a GitHub Issue with your reproducer.

**Comment by melloware** [comment]
Hmmm it doesn't seem to be happening on the showcase: https://primereact.org/multiselect/#filter

**Reply by mohamadbdeir** [reply]
It does work fine when the list of options is static, in my case, the options are being dynamically fetched from rest API on filter so at some point, a selected value may not be available in the array of options and this is when it shows as null.

**Reply by melloware** [reply]
Oh i see yeah if your items have been removed from the list I can see what that is a problem if you are constantly swapping the list out from underneath the component.

**Reply by mohamadbdeir** [reply]
I was thinking that I can try to fix that by contributing to the project, the rules said to discuss it before, hence this discussion.

**Reply by melloware** [reply]
Oh yeah just opena GitHub ticket and a PR. The community is very welcoming

**Reply by mertsincan** [reply]
Welcome! yes, could you please create a Github issue for this bug?
